README.md

```diff
@@ -208,40 +397,89 @@ pipeline core process:
 Example: 
     pipe-gaps -c config/sample-from-file-1.json --threshold 1.3
 ```
-### Apache Beam integration
 
-Just use `--pipe-type beam` option:
-```
-pipe-gaps --pipe-type beam -c config/sample-from-file-1.json --threshold 1.3
-```
-This will run by default with DirectRunner.
-To run on DataFlow, add `--runner dataflow` option.
+> [!NOTE]
+> Any option passed to the CLI not explicitly documented will be inside "options" key of the configuration
+  (see above). 
 
-Beam integrated pipeline will parallelize grouping inputs by SSVID and year.
+## Implementation details
 
-### Handle of open gaps
+The pipeline is implemented over a (mostly) generic structure that supports:
+1. Grouping all _main inputs_ by some composite key with **SSVID**
+    and a time interval (**TI**). For example, you can group by **(SSVID, YEAR)**.
+2. Grouping _side inputs_ by **SSVID**.
+3. Grouping _boundaries_ (first and last **AIS** messages of each group) by **SSVID**.
+4. Processing _main inputs_ groups from 1.
+5. Processing _boundaries_ from 4 together with _side inputs from_ 2, both grouped by **SSVID**.
 
-When an open gap is closed,
-a new gap event is created. 
-Two gap events will exist with the same `gap_id`,
-one will be an old open gap, and the other one will be the current closed active gap.
+Below there is a [diagram](#flow-chart) that describes this work flow.
 
-To query all active gaps,
-you will just need to query the last versions for every `gap_id`.
-For example:
-```sql
-SELECT *
-    FROM (
-      SELECT
-          *,
-          MAX(gap_version)
-              OVER (PARTITION BY gap_id)
-              AS last_version,
-      FROM `world-fishing-827.scratch_tomas_ttl30d.pipe_ais_gaps_filter_no_overlapping_and_short`
-    )
-    WHERE gap_version = last_version
-```
+> [!TIP]
+> In the case of the Apache Beam integration with [Dataflow runner],
+  the groups can be processed in parallel accross different workers.
+
+
+### :warning: Important note on grouping input AIS messages by `ssvid` 
 
+<div align="justify">
+
+The gap detection pipeline fetches **AIS** position messages,
+filtering them by `good_seg`, and groups them by `ssvid`.
+Since **we know** different vessels can broadcast with the same `ssvid`,
+this can potentially lead to the situation in which we have a gap
+with an `OFF` message from one vessel
+and a `ON` message from another vessel (or viceversa).
+This can happen because currently,
+the gap detection process just orders
+all the messages from the same `ssvid` by `timestamp`
+and then evaluates each pair of consecuive messages.
+In consequence,
+it will just pick the last `OFF` (or the first `ON`)
+message in the chain when constructing a gap,
+and we could have have "inconsistent" gaps
+in the sense we described above.
+We believe those will be a very small
+amount of the total gaps,
+and we aim in the future to find a solution to this problem.
+One option could be to use something more stable like `vessel_id`
```

Comment:
Actually, vessel_id is fairly unstable (at least compared to ssvid). So let's use a more accurate phrasing. E.g. "One option could be to use e.g. `vessel_id` which has a much higher chance of separating messages from different vessels that broadcast under the same ssvid" (even though this will still lead to loads of mismatches, unless we improve vessel_id A LOT).

Reply:
Thanks! I will change it.